core/src/main/java/org/elasticsearch/search/sort/SortBuilder.java

```diff
@@ -55,34 +54,24 @@
     protected SortOrder order = SortOrder.ASC;
     public static final ParseField ORDER_FIELD = new ParseField("order");
 
-    private static final Map<String, SortBuilder<?>> PARSERS;
+    private static final Map<String, Parser<?>> PARSERS;
+
+    @FunctionalInterface
+    public interface Parser<T extends SortBuilder<?>> {
```

Comment:
I wonder if we should simply use a BiFunction instead of this parser? Maybe I have to get more used to this functional stuff but it feels weird to me that we declare this parser interface. One more thing, does it need to be public?